### PR TITLE
feat: add ElementHandle.hasFocus() method

### LINKED ISSUE #2159: [Feature] hasFocus method on element
I didn't see this in the docs, but I may not have looked in the right place. Is it possible to check if an element has focus?

### PR DESCRIPTION
**Pull request by aslushnikov**
Note: `page.hasFocus()` would look weird since it clashes with Web's
`document.hasFocus()`.

Fixes #2159

### CODE REVIEW

File: src/dom.ts [hunk 1]
```diff
@@ -157,6 +157,15 @@ export class ElementHandle<T extends Node = Node> extends js.JSHandle<T> {
     }, {});
   }
 
+  async hasFocus(): Promise<boolean> {
+    return this._evaluateInUtility(({node}) => {
+      if (node.nodeType !== Node.ELEMENT_NODE)
+        throw new Error('Not an element');
+      const element = node as unknown as Element;
+      return element === document.activeElement;
```
Comment (JoelEinbinder) [inline src/dom.ts:165]: This doesn't work with shadow dom.

Comment (aslushnikov) [reply src/dom.ts:165]: Good catch!

**Comment by JoelEinbinder** [comment]
I think this should be `isFocused` not `hasFocus`. To me, `hasFocus` means that an element or one of its descendants is focused.

That being said, maybe hasFocus is more useful?

**Comment by aslushnikov** [comment]
There's also `isFocusWithin` to match css pseudo-class?

**Comment by aslushnikov** [comment]
Discussed offline and decided not to land this

**Pull request closed without merging**